controllers/storagecluster/consumer_grpc_error_handling.go

```diff
@@ -0,0 +1,184 @@
+package storagecluster
+
+import (
+	"time"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+)
+
+var (
+	// ensureCreated funcs can return only error or nil
+	// if it returns nil Reconcile go ahead and call other funcs
+	// But if it returns error Reconcile will get requeued, We can not determine
+	// if the error is resolvable or not either we need to requeue of not
+	// isErrorResolvable will help us for the same if isErrorResolvable[instance.UID] == false we won't requeue
+	isErrorResolvable = map[types.UID]bool{}
+
+	// requeueAfter holds info about requeue time
+	requeueAfter = map[types.UID]time.Duration{}
+)
+
+// handleOnBoardConsumerErrors returns the bool which tells to requeue or not
+func (r *StorageClusterReconciler) handleOnboardConsumerErrors(instance *ocsv1.StorageCluster, err error) error {
+
+	if err == nil {
+		return nil
+	}
+
+	s, ok := status.FromError(err)
+	if !ok {
+		// This is not gRPC error. The operation must have failed before gRPC
+		// method was called, otherwise we would get gRPC error.
+		return err
+	}
+
+	// internal error, do requeue
+	if s.Code() == codes.Internal {
+		return err
+	}
+
+	// token not valid, do not requeue
+	if s.Code() == codes.InvalidArgument {
+		reason := "Token is invalid verify the token again or contact provider admin"
+		r.Log.Error(err, "ocs-to-ocs:onboard: "+reason)
```

Comment:
I don't think we want to say `ocs-to-ocs` in logs.

Reply:
Actually, I want to differentiate b/w logs of ocs-to-ocs that's why I did this. maybe we can think of some other keyword.